dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/config/DimensionDataCloudControlComputeServiceContextModule.java

```diff
@@ -0,0 +1,254 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.dimensiondata.cloudcontrol.config;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Predicate;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.name.Named;
+import org.jclouds.compute.reference.ComputeServiceConstants;
+import org.jclouds.dimensiondata.cloudcontrol.DimensionDataCloudControlApi;
+import org.jclouds.dimensiondata.cloudcontrol.domain.NetworkDomain;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Server;
+import org.jclouds.dimensiondata.cloudcontrol.domain.State;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Vlan;
+import org.jclouds.dimensiondata.cloudcontrol.domain.VmTools;
+import org.jclouds.dimensiondata.cloudcontrol.features.NetworkApi;
+import org.jclouds.dimensiondata.cloudcontrol.features.ServerApi;
+import org.jclouds.logging.Logger;
+
+import javax.annotation.Resource;
+import java.text.MessageFormat;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.jclouds.util.Predicates2.retry;
+
+public class DimensionDataCloudControlComputeServiceContextModule extends AbstractModule {
+
+   @Resource
+   private Logger logger = Logger.NULL;
+
+   public static final String VLAN_DELETED_PREDICATE = "VLAN_DELETED_PREDICATE";
+   public static final String VLAN_NORMAL_PREDICATE = "VLAN_NORMAL_PREDICATE";
+   public static final String NETWORK_DOMAIN_DELETED_PREDICATE = "NETWORK_DOMAIN_DELETED_PREDICATE";
+   public static final String NETWORK_DOMAIN_NORMAL_PREDICATE = "NETWORK_DOMAIN_NORMAL_PREDICATE";
+   public static final String SERVER_STARTED_PREDICATE = "SERVER_STARTED_PREDICATE";
+   public static final String SERVER_STOPPED_PREDICATE = "SERVER_STOPPED_PREDICATE";
+   public static final String SERVER_DELETED_PREDICATE = "SERVER_DELETED_PREDICATE";
+   public static final String SERVER_NORMAL_PREDICATE = "SERVER_NORMAL_PREDICATE";
+   public static final String VM_TOOLS_RUNNING_PREDICATE = "VM_TOOLS_RUNNING_PREDICATE";
+
+   @Override
+   protected void configure() {
+
+   }
+
+   @Provides
+   @Named(VLAN_DELETED_PREDICATE)
+   protected Predicate<String> provideVlanDeletedPredicate(final DimensionDataCloudControlApi api,
+         final ComputeServiceConstants.Timeouts timeouts, final ComputeServiceConstants.PollPeriod pollPeriod) {
+      return retry(new VlanState(api.getNetworkApi(), State.DELETED), timeouts.imageAvailable,
```

Comment:
The `imageAvailable` timeout is meant to be used by the `ImageExtension` when waiting for a new image to be created. Instead of using it, just create a property such as `operation-timeout` that can be configured as a generic timeout for DimensionDataOperations, and get it injected here and in other methods that don't have a matching timeout property in the `ComputeServiceConstants.Timeouts` class.